streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -272,11 +287,20 @@ private void commitOffsets() {
                 consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
                 stateMgr.putOffsetLimit(partition, offset);
             }
-            try {
-                consumer.commitSync(consumedOffsetsAndMetadata);
-            } catch (final CommitFailedException cfe) {
-                log.warn("{} Failed offset commits: {} ", logPrefix, consumedOffsetsAndMetadata);
-                throw cfe;
+
+            if (exactlyOnceEnabled) {
+                producer.sendOffsetsToTransaction(consumedOffsetsAndMetadata, applicationId);
+                producer.commitTransaction();
```

Comment:
This can only fail with `ProducerFencedException` that is handled by the caller (ie, `StreamThread`).